lib/src/adaptative_builder_widget.dart

```diff
@@ -0,0 +1,92 @@
+import 'package:flutter/widgets.dart';
+
+/// A function that creates an object of type [T].
+typedef ValueBuilder<T> = T Function(BuildContext context);
+
+abstract class AdaptativeBuilderWidget<T, B> extends StatefulWidget {
+  const AdaptativeBuilderWidget({
+    Key key,
+    @required this.builder,
```

Comment:
filip mentioned it's preferred to put 'required' arguments before others

Reply:
Is this really what's wanted for widgets?

I saw somewhere else a statement where keys should always be first and children last – and a quick look through Flutter seems to confirm this (with some exceptions).

Reply:
No big deal in any case. There's a lint for this, which would fire for the code above, but that's obviously optional. Consider it a nit and go with whatever works best for this codebase.